service/src/main/java/skills/storage/repos/UserAchievedLevelRepo.groovy

```diff
@@ -526,6 +534,24 @@ interface UserAchievedLevelRepo extends CrudRepository<UserAchievement, Integer>
             @Param("containerType") SkillDef.ContainerType containerType
     )
 
+    @Query('''select ua.skillId as skillId, ua.level as level, count(ua.id) as numberUsers, ut.value as userTag 
```

Comment:
- I don't believe there is a need to join with SkillDef, pure usage of projectId+skillId is enough and containterType param can then be removed; 
- could add ``ua.level is not null`` for an extra safety net